runtime/src/iree/base/internal/time.c

```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/base/internal/time.h"
+
+#include <time.h>
+
+#include "iree/base/target_platform.h"
+
+IREE_API_EXPORT iree_time_t iree_time_now_impl(void) {
```

Comment:
> not a fan of the duplication - I'd say add an iree/base/internal/time.h/.c, put the function in there as `int64_t iree_time_now_ns(void)`, call it from the iree/base/time.c `iree_time_now` and also from the console tracing provider.

Oh, is this fine, or do you want it to return `int64_t` and not depend on `iree::base::core_headers`?